server/src/test/java/org/apache/gravitino/server/web/TestVersioningFilter.java

```diff
@@ -259,4 +259,35 @@ public void testGetHeaderNames() {
     assertTrue(actualHeaderNames.contains("Header2"));
     assertTrue(actualHeaderNames.contains("CustomHeader"));
   }
+
+  @Test
+  public void testDoFilterWithHeaderContainingValidVersionAsSubstring() throws Exception {
+    VersioningFilter filter = new VersioningFilter();
+    FilterChain mockChain = mock(FilterChain.class);
+    HttpServletRequest mockRequest = mock(HttpServletRequest.class);
+    HttpServletResponse mockResponse = mock(HttpServletResponse.class);
+    when(mockRequest.getHeaders("Accept"))
+        .thenReturn(
+            new Vector<>(Collections.singletonList("foo application/vnd.gravitino.v1+json bar"))
+                .elements(),
+            new Vector<>(Collections.singletonList("foo application/vnd.gravitino.v2+json bar"))
+                .elements(),
+            new Vector<>(Collections.singletonList("foo application/vnd.gravitino.v3+json bar"))
+                .elements());
+
+    filter.doFilter(mockRequest, mockResponse, mockChain);
+    verify(mockChain).doFilter(any(), any());
+
+    reset(mockChain, mockResponse);
+
+    filter.doFilter(mockRequest, mockResponse, mockChain);
+    verify(mockChain, never()).doFilter(any(), any());
+    verify(mockResponse).sendError(HttpServletResponse.SC_NOT_ACCEPTABLE, "Unsupported version");
+
+    reset(mockChain, mockResponse);
+
+    filter.doFilter(mockRequest, mockResponse, mockChain);
+    verify(mockChain, never()).doFilter(any(), any());
+    verify(mockResponse).sendError(HttpServletResponse.SC_NOT_ACCEPTABLE, "Unsupported version");
+  }
```

Comment:
This test does not test multiple header values or parameters at all. Instead, it creates one header string at a time. It needs to test multiple values, e.g., "application/vnd.gravitino.v1+json, application/json" or "application/vnd.gravitino.v1+json; q=0.9," which is the main point of https://github.com/apache/gravitino/issues/8053.